scirpy/_tools/_clonotype_imbalance.py

```diff
@@ -0,0 +1,294 @@
+    additional_hue
+        An additional grouping factor. If the `case_label` was tumor for example, this could
+        help make a distinction between imbalance in lung and colorectal tumors.
+    case_label
+        The label in `groupby` column that we want to compare. 
+    control_label
+        The label in `groupby` column that we use as a baseline for comparison. If not set
+        (None by default), all labels that are not equal to `case_label` make up the baseline. 
+    target_col
+        The clusters (clonotypes by default) that are imbalanced. 
+
+    Returns
+    -------
+    A list, where each item consist of a hue, list of cases, list of controls,
+    number of cases, number of controls. 
+    """
+
+    case_control_groups = []
+    group_cols = [groupby, replicate_col]
+    if additional_hue is None:
+        hues = [None]
+    else:
+        group_cols.append(additional_hue)
+        hues = df[additional_hue].unique()
+    df = df.groupby(group_cols, observed=True).agg("size").reset_index()
+
+    for hue in hues:
+        if hue is None:
+            tdf = df
+        else:
+            tdf = df.loc[df[additional_hue] == hue, :]
+        cases = tdf.loc[df[groupby] == case_label, :]
+        ncase = cases[0]
+        cases = cases[replicate_col]
+        if control_label is None:
+            controls = tdf.loc[df[groupby] != case_label, :]
+        else:
+            controls = tdf.loc[df[groupby] == control_label, :]
+        ncontrol = controls[0]
+        controls = controls[replicate_col]
+        case_control_groups.append([hue, cases, controls, ncase, ncontrol])
+
+    return case_control_groups
+
+
+def _calculate_imbalance(
+    case_sizes: Union[List, np.ndarray, pd.Series],
+    control_sizes: Union[List, np.ndarray, pd.Series],
+    ncase: Union[List, np.ndarray, pd.Series],
+    ncontrol: Union[List, np.ndarray, pd.Series],
+) -> Tuple[float, float, np.ndarray, np.ndarray]:
+    """Calculate statistics for the probability of an imbalance in the contingency table among two groups. 
+    
+    Parameters
+    ----------
+    case_sizes
+        An iterable of sizes (conts or normalized counts) in a given group for each replicates. 
+    control_sizes
+        An iterable of sizes (conts or normalized counts) in the control group for each replicates.
+    ncase
+        Total size (all counts or sum of normalized counts) of a given group for each replicates. 
+    ncontrol
+        Total size (all counts or sum of normalized counts) of the control group for each replicates.
+
+    Returns
+    -------
+    The p-value of a Fischers exact test and a logFoldChange of the case frequency compared to the control frequency
+    and the relative sizes for case and control groups. 
+    """
+
+    rel_case_sizes = case_sizes / np.array(ncase)
+    rel_control_sizes = control_sizes / np.array(ncontrol)
+    case_mean_freq = np.mean((case_sizes + 0.0001) / np.array(ncase))
```

Comment:
What's the point of the pseudo-frequency here? 
Would be good to add a comment either here or to the function documentation.

Reply:
Moved this to logFC calculation